client/block-builder/src/lib.rs

```diff
@@ -212,6 +220,23 @@ where
 			proof,
 		})
 	}
+
+	/// Create the inherents for the block.
+	///
+	/// Returns the inherents created by the runtime or an error if something failed.
+	pub fn create_inherents(
+		&mut self,
+		inherent_data: sp_inherents::InherentData,
+	) -> Result<Vec<Block::Extrinsic>, ApiErrorFor<A, Block>> {
+		let block_id = self.block_id;
+		self.api.execute_in_transaction(move |api| {
+			TransactionOutcome::Rollback(api.inherent_extrinsics_with_context(
```

Comment:
Nit: Could we leave a comment that `create_inherents` are not supposed to change any state, since a unconditional rollback looks a bit unusual and will evoke questions.